extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -84,10 +87,27 @@ require([
 	});
 
 	// fetch data for all recirculation modules
-	// TODO lazy load some data on scroll
 	liftigniter.fetch('ns');
-	liftigniter.fetch('wiki');
-	discussions.fetch();
+
+	window.addEventListener('scroll', function lazyLoadHandler() {
+		($.throttle(50, function () {
+			if(!wikiArticlesFetched) {
+				var mcfOffset = $mixedContentFooter.offset().top,
+					scrollPosition = $(window).scrollTop(),
+					windowInnerHeight = $(window).height(),
+					lazyLoadOffset = 500,
+					aproachingMCF = scrollPosition > mcfOffset - windowInnerHeight - lazyLoadOffset;
+
+				if (aproachingMCF) {
+					liftigniter.fetch('wiki');
+					discussions.fetch();
+					wikiArticlesFetched = true;
+					window.removeEventListener('scroll', lazyLoadHandler);
+				}
+			}
+		}))()
+	});
+
 
 	// TODO handle errors
 	// TODO LI tracking
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)